subsys/bluetooth/host_extensions/conn_evt_cb.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/hci.h>
+#include <bluetooth/evt_cb.h>
+#include <helpers/nrfx_gppi.h>
+#include <bluetooth/hci_vs_sdc.h>
+
+#include <nrfx_egu.h>
+
+#if defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU0)
+#define EGU_INST_IDX 0
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU1)
+#define EGU_INST_IDX 1
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU2)
+#define EGU_INST_IDX 2
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU3)
+#define EGU_INST_IDX 3
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU4)
+#define EGU_INST_IDX 4
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU5)
+#define EGU_INST_IDX 5
+#elif defined(CONFIG_BT_CONN_EVT_CB_EGU_INST_EGU020)
+#define EGU_INST_IDX 020
+#else
+#error "Unknown EGU instance"
+#endif
+
+static nrfx_egu_t egu_inst = NRFX_EGU_INSTANCE(EGU_INST_IDX);
+
+struct conn_evt_cb_cfg {
+	struct bt_conn_evt_cb cb;
+	struct bt_conn *conn;
+	void *user_data;
+	uint32_t prepare_time_us;
+
+	uint8_t ppi_channel;
+	uint16_t conn_interval_us;
+};
+
+static struct conn_evt_cb_cfg cfg[CONFIG_BT_MAX_CONN];
+static struct k_work_delayable work_queue_items[CONFIG_BT_MAX_CONN];
+
+static void work_handler(struct k_work *work)
+{
+	uint8_t index = ARRAY_INDEX(work_queue_items, work);
+
+	cfg[index].cb.prepare(cfg[index].conn, cfg[index].user_data);
+
+	/* Schedule the callback in the connection event does not occur because
+	 * of peripheral latency or scheduling conflicts.
+	 */
+	k_work_reschedule(&work_queue_items[index], K_USEC(cfg[index].prepare_time_us));
```

Comment:
```suggestion
	k_work_reschedule(&work_queue_items[index], K_USEC(cfg[index].conn_interval_us));
```

We should maybe also record the time of the `SDC_HCI_VS_CONN_EVENT_TRIGGER_ROLE_CONN` event and use that as anchor? I worry about scheduling delays.

Reply:
We could store the time of the trigger, but that will require some additional HW resources. We have some plans on how to expose this information. That will allow us to simplify the implementation.

I would assume that the current implementation is good enough to start out with. It is anyways marked with "experimental".

Reply:
I was thinking of capturing `k_cycle_get_32` in the EGU ISR. But I agree it may not be necessary.

Obs: The suggestion above in this thread point out a bug, I think. Do you agree the "backup plan" reschedule should be a connection interval in the future?

Reply:
Yes, it should be a connection interval!